app/src/main/java/com/blockeq/stellarwallet/activities/WalletActivity.kt

```diff
@@ -158,10 +169,19 @@ class WalletActivity : BaseActivity(), OnLoadAccount, OnLoadEffects, KeyboardUti
         }
     }
 
-    override fun onLoadEffects(result: ArrayList<EffectResponse>?) {
-        val fragment = supportFragmentManager.findFragmentByTag(WalletFragmentType.WALLET.name)
-        if (fragment != null) {
-            (fragment as WalletFragment).onLoadEffects(result)
-        }
+//    override fun onLoadEffects(result: ArrayList<EffectResponse>?) {
+//        val fragment = supportFragmentManager.findFragmentByTag(WalletFragmentType.WALLET.name)
+//        if (fragment != null) {
+//            (fragment as WalletFragment).onLoadEffects(result)
+//        }
+//    }
+
+    private fun subscribeDataStreams(viewModel : EffectsViewModel) {
+        viewModel.getEffectsList().observe(this, Observer {
+            val fragment = supportFragmentManager.findFragmentByTag(WalletFragmentType.WALLET.name)
+            if (fragment != null) {
+                (fragment as WalletFragment).onLoadEffects(it)
```

Comment:
fragment may have the view destroyed and be in the memory waiting to be attached.